pkg/volume/glusterfs/glusterfs.go

```diff
@@ -479,22 +510,127 @@ func (d *glusterfsVolumeDeleter) Delete() error {
 	return nil
 }
 
+func (r *glusterfsVolumeProvisioner) ConstructBitmap() error {
+
+	pvList, err := r.plugin.host.GetKubeClient().Core().PersistentVolumes().List(v1.ListOptions{LabelSelector: labels.Everything().String()})
+	if err != nil {
+		glog.Errorf("glusterfs: failed to get existing persistent volumes")
+		return fmt.Errorf("glusterfs: failed to get existing persistent volumes")
+	}
+	// Collect the SC names and allocated GIDs from existing PVs.
+	scNameGids := make(map[string][]string)
+	for _, pv := range pvList.Items {
+		if pv.Annotations[annDynamicallyProvisioned] == glusterfsPluginName {
+			scNameInAnnot := pv.Annotations[annStorageClass]
+			scNameGids[scNameInAnnot] = append(scNameGids[scNameInAnnot], pv.Annotations[volumehelper.VolumeGidAnnotationKey])
+		}
+	}
+	glog.V(3).Infof("glusterfs: PVs with SC name and allocated GID Map :%v ", scNameGids)
+
+	for key := range scNameGids {
+		scdetails, _ := r.plugin.host.GetKubeClient().Storage().StorageClasses().Get(key)
+		if scdetails.Parameters["gidMin"] == "" {
+			scdetails.Parameters["gidMin"] = defaultGidMin
```

Comment:
storageclass params can be changed... need to handle gid min/max updating after the initial load